bash_completion

```diff
@@ -2078,14 +2078,25 @@ _command_offset()
                 # complete -F <function>
 
                 # get function name
-                local func=${cspec#*-F }
-                func=${func%% *}
+                local func=${cspec#*-F } newcspec
 
-                if ((${#COMP_WORDS[@]} >= 2)); then
-                    $func "$cmd" "${COMP_WORDS[-1]}" "${COMP_WORDS[-2]}"
-                else
-                    $func "$cmd" "${COMP_WORDS[-1]}"
-                fi
+                for ((i = 0; i <= 1; i++)); do
+                    func=${func%% *}
+
+                    if ((${#COMP_WORDS[@]} >= 2)); then
+                        $func "$cmd" "${COMP_WORDS[-1]}" "${COMP_WORDS[-2]}"
+                    else
+                        $func "$cmd" "${COMP_WORDS[-1]}"
+                    fi
+
+                    # restart completion (once) if function exited with 124
+                    if [[ $? == 124 ]]; then
+                        newcspec=$(complete -p "$compcmd" 2>/dev/null)
+                        func=${newcspec#*-F }
```

Comment:
- What happens when `-F func` is not included in the new completion spec? I think `complete --p ...` will be executed in the second loop.
- What happens when the completion spec is removed, i.e., `newcspec` becomes empty? `$cmd` will be executed unintendedly in the second loop.

I think you may also include the checks `[[ -n $cspec ]]` and `[[ ${cspec#* -F } != "$cspec" ]]` inside the loop.

Reply:
I think we could just run the existing code which processes the original compspec again to handle these edge cases. In my latest commit I do this by increasing the scope of the for-loop.

Reply:
Thanks! I have pushed an additional modification b2409562 in which I moved the fallback for the empty `cspec` also into the loop.